### ISSUE TITLE
Taking diff generation out of the HTTP request cycle

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#11200]
## How it is
By default we display the diff if you [look a submit-request in the UI](https://build.opensuse.org/request/show/670076). If the backend has not gotten around to generate the diff yet, we trigger the generation in the HTTP request cycle. This has several drawbacks:

## Consequences
People have to wait sometimes a *very long* time until the page renders:
![Response Time Last 24 hours as of this writing](https://user-images.githubusercontent.com/514785/120492930-ccaaa180-c3ba-11eb-900a-da12daa4a736.png)

There are several pages that list many submit-requests (like the [revisions page](https://build.opensuse.org/package/revisions/openSUSE:Factory/chromium)).  This frequently becomes problem as soon as some spider-bot (that doesn't respect robots.txt and/or is not identifiable) runs over those because diff generation overloads the backends request slots.

Currently we are also triggering a [delayed job as soon as a submit-request is created that is requesting the diff](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/jobs/bs_request_action_webui_infos_job.rb), so we don't run too often into the situation that a diff isn't generated when people look at requests.

## How it should be
If people look at a submit-request and the diff isn't generated yet we should tell them and offer an option to refresh.

## Proposal
Take the diff generation out of the HTTP request cycle

- User is hitting /request/show/1234
- The frontend asks the backend for the diff with the `cacheonly=1` option
- If the backend has the diff we render it
- If the backend doesn't have the diff the frontend 
  - renders an explanation and a refresh button
  - schedules a background job to request the diff without the `cacheonly` option
- the delayed job queue runs only N of those jobs concurrently (not sure what N should be yet)
- stop scheduling the [BsRequestActionWebuiInfosJob](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/jobs/bs_request_action_webui_infos_job.rb)

This way HTTP response performance would go up, people would know what's going on, have a way to do something about it (refresh) and spiders have no way of overloading the backend with diff requests anymore which would bring up reliability.

**Comment by lrupp** [comment]
Just a question: will this change affect API calls (mainly asking for `osc` reviews) or just the WebUI ?

**Comment by hennevogel** [comment]
Nope, API calls are always authenticated and osc doesn't even display the diff by default. So nothing will change.
